hayeon/programmers-2/스킬트리.js

```diff
@@ -0,0 +1,28 @@
+function solution(skill, skill_trees) {
+  let result = 0;
+
+  const filteredTrees = skill_trees.map((tree) => {
+    return tree.split("").filter((el) => skill.includes(el));
+  });
+
+  for (let i = 0; i < filteredTrees.length; i++) {
+    let isValid = true;
+
+    for (let j = 0; j < filteredTrees[i].length; j++) {
+      if (skill[j] !== filteredTrees[i][j]) {
+        isValid = false;
+        break;
+      }
+    }
+    if (isValid) result++;
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(skill, skill_trees) {
  let result = 0;

  const filteredSkillTrees = skill_trees.map((tree) => {
    return tree.split("").filter((el) => skill.includes(el));
  });

  filteredSkillTrees.forEach((skillTrees) => {
    let isValid = true;

    skillTrees.forEach((tree, i) => {
      if (skill[i] !== tree) isValid = false;
    });
      
    if (isValid) result++;
  });

  return result;
}
```
- 변수명 일부 변경 및 함수형으로 작성해봤습니다! 한번 참고해주세요
- 아래는 정규표현식을 이용한 풀이입니다!
```js
function solution(skill, skill_trees) {
  const regex = new RegExp(`[^${skill}]`, "g");
  let result = 0;

  skill_trees.forEach((target) => {
    const replacedSkill = target.replace(regex, "");
    let isValid = true;

    for (let i = 0; i < replacedSkill.length; i++) {
      if (replacedSkill[i] !== skill[i]) {
        isValid = false;
      }
    }

    isValid && result++;
  });

  return result;
}
```

Reply:
- 오 forEach로 쓰니까 훨씬 더 깔끔해보이네요! 
- 문제풀이 보면서 정규표현식으로 푸는 풀이도 봤었는데 이 방법으로도 다시 풀어보겠습니다!! 
- 정규표현식으로 푸는 법은 아직 익숙하지 않아서 어렵네유..🥲